### PR TITLE
[BEAM-4775] Add GetJobMetrics RPC.

### PR DESCRIPTION
**Pull request by robertwb**
This just adds the RPC to the Jobs API.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @ryan-williams @ajamato

File: model/job-management/src/main/proto/beam_job_api.proto [hunk 1]
```diff
@@ -179,6 +183,17 @@ message JobState {
 }
 
 
+message GetJobMetricsRequest {
+  string job_id = 1; // (required)
+}
+
+// All metrics for a given job.  Runners may support one or the other or both.
+message GetJobMetricsResponse {
```
Comment (ryan-williams) [inline model/job-management/src/main/proto/beam_job_api.proto]: I started with this, per https://s.apache.org/get-metrics-api.

Converting between this proto and the SDK reprs ([Java](https://github.com/apache/beam/pull/7934/files#diff-e722f484e12b515353e20532d3550465), [Python](https://github.com/apache/beam/pull/7934/files#diff-08a37e2757c9df3f0e1d0ba3e2121eecR328)) made it clear that we were actually introducing a proto representation of SDKs' `MetricResults`.

For example, `MetricResultsProtos.java` was originally `GetJobMetricsResponseUtils.java`:

```
public class GetJobMetricsResponseUtils {
  public static GetJobMetricsResponse   toProto(        MetricResults metricResults) { … }
  public static         MetricResults fromProto(GetJobMetricsResponse          resp) { … }
}
```

The signatures (and implementations) implied a broader purpose in ongoing discussions around metrics structures in the protos and SDKs. 

Framing it as just part of this RPC obfuscated that role, to me.

Comment (robertwb) [reply model/job-management/src/main/proto/beam_job_api.proto]: OK, re-introduced the proto.

Comment (ryan-williams) [reply model/job-management/src/main/proto/beam_job_api.proto]: Thanks. Should it live in `metrics.proto` though?

File: model/job-management/src/main/proto/beam_job_api.proto [hunk 2]
```diff
@@ -179,6 +183,21 @@ message JobState {
...
 
+message GetJobMetricsRequest {
+  string job_id = 1; // (required)
+}
+
+message GetJobMetricsResponse {
+  MetricsResults metrics = 1;
+}
+
+// All metrics for a given job.  Runners may support one or the other or both.
```
Comment (ryan-williams) [inline model/job-management/src/main/proto/beam_job_api.proto]: Today, runners either populate ["attempted" only] or both; dunno if worth clarifying here.

File: model/job-management/src/main/proto/beam_job_api.proto (hunk 1 above)
Comment (robertwb) [reply model/job-management/src/main/proto/beam_job_api.proto]: It is really specific to a particular job, but I don't feel that strongly.

File: model/job-management/src/main/proto/beam_job_api.proto (hunk 2 above)
Comment (robertwb) [reply model/job-management/src/main/proto/beam_job_api.proto]: It's acceptable to populate committed only as well.

File: model/job-management/src/main/proto/beam_job_api.proto (hunk 1 above)
Comment (ryan-williams) [reply model/job-management/src/main/proto/beam_job_api.proto]: sounds good

File: model/job-management/src/main/proto/beam_job_api.proto [hunk 3]
```diff
@@ -179,6 +183,21 @@ message JobState {
...
+message MetricsResults {
+  org.apache.beam.model.pipeline.v1.MonitoringInfo attempted = 1;
```
Comment (ryan-williams) [inline model/job-management/src/main/proto/beam_job_api.proto]: eh, these were supposed to be `repeated`, I think

**Pull request merged**